src/main/java/net/javadiscord/javabot/data/h2db/message_cache/MessageCache.java

```diff
@@ -0,0 +1,211 @@
+package net.javadiscord.javabot.data.h2db.message_cache;
+
+import lombok.extern.slf4j.Slf4j;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.*;
+import net.dv8tion.jda.api.events.message.MessageDeleteEvent;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.events.message.MessageUpdateEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import net.dv8tion.jda.api.interactions.components.buttons.Button;
+import net.dv8tion.jda.api.requests.restaction.MessageAction;
+import net.javadiscord.javabot.Bot;
+import net.javadiscord.javabot.data.config.guild.MessageCacheConfig;
+import net.javadiscord.javabot.data.h2db.DbHelper;
+import net.javadiscord.javabot.data.h2db.message_cache.dao.MessageCacheRepository;
+import net.javadiscord.javabot.data.h2db.message_cache.model.CachedMessage;
+import net.javadiscord.javabot.systems.commands.IdCalculatorCommand;
+import net.javadiscord.javabot.util.GuildUtils;
+import net.javadiscord.javabot.util.TimeUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.sql.SQLException;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Listens for Incoming Messages and stores them in the Message Cache.
+ */
+@Slf4j
+public class MessageCache extends ListenerAdapter {
+	List<CachedMessage> cache = new ArrayList<>();
+	/**
+	 * Amount of messages since the last synchronization.
+	 * <p>
+	 * If a certain threshold is reached, messages will be synchronized to reduce the chances of loosing
+	 * messages during an unexpected shutdown.
+	 */
+	private int messageCount = 0;
+
+	/**
+	 * Creates a new messages & loads messages from the DB into a List.
+	 */
+	public MessageCache() {
+		try {
+			cache = new MessageCacheRepository(Bot.dataSource.getConnection()).getAll();
+		} catch (SQLException e) {
+			log.error("Something went wrong during retrieval of stored messages.");
+		}
+	}
```

Comment:
use `try-with-resources` to automatically free up resources
```suggestion
		try (Connection con = Bot.dataSource.getConnection()){
			cache = new MessageCacheRepository(con).getAll();
		} catch (SQLException e) {
			log.error("Something went wrong during retrieval of stored messages.");
		}
	}
```